src/cpp/multisolver_interface/SolverFactory.cpp

```diff
@@ -9,22 +11,48 @@
 #include "antares-xpansion/multisolver_interface/SolverConfig.h"
 #include "antares-xpansion/multisolver_interface/SolverFactory.h"
 #include "antares-xpansion/xpansion_interfaces/LogUtils.h"
-std::vector<std::string> available_solvers;
 
-std::vector<std::string> SolverLoader::GetAvailableSolvers(
-    std::shared_ptr<ILoggerXpansion> logger) {
+namespace {
+// not std::string, static initialization fiasco with tests static
+// initialization
+const char *COIN_STR("COIN");
+const char *CBC_STR("CBC");
+const char *CLP_STR("CLP");
+const char *XPRESS_STR("XPRESS");
+
+#include "antares-xpansion/xpansion_interfaces/LogUtils.h"
+std::vector<std::string> available_solvers;
+std::once_flag solver_flag;
+void GetAvailableSolversInternal(std::shared_ptr<ILoggerXpansion> logger) {
   if (available_solvers.empty()) {
-    LoadXpress::XpressLoader xpress_loader(logger);
+    LoadXpress::XpressLoader xpress_loader(std::move(logger));
     if (xpress_loader.XpressIsCorrectlyInstalled(true)) {
-      available_solvers.push_back(XPRESS_STR);
+      available_solvers.emplace_back(XPRESS_STR);
     }
 #ifdef COIN_OR
-    available_solvers.push_back(CLP_STR);
-    available_solvers.push_back(CBC_STR);
+    available_solvers.emplace_back(CLP_STR);
+    available_solvers.emplace_back(CBC_STR);
 #endif
   }
+}
+}  // namespace
+
+std::vector<std::string> SolverLoader::GetAvailableSolvers(
+    std::shared_ptr<ILoggerXpansion> logger) {
+  std::call_once(solver_flag, GetAvailableSolversInternal, logger);
   return available_solvers;
 }
+std::vector<std::string> SolverLoader::GetSupportedSolvers() {
+  static std::vector<std::string> supported_solvers;
+  if (supported_solvers.empty()) {
+    supported_solvers.emplace_back(XPRESS_STR);
```

Comment:
I don't understand what happens here, it looks like Xpress will always be in the supported_solver list

Reply:
Ok, I'm adding a comment. Support is "I know such solver exists" it differs from "available" because it doesn't check for licence for example.
Currently the only usage is for testing (buty tricky to move in tests)